tests/framework/builder.py

```diff
@@ -4,29 +4,45 @@
 
 import json
 import os
+import tempfile
+from pathlib import Path
 from conftest import init_microvm
-from framework.artifacts import Artifact, DiskArtifact
+from framework.artifacts import Artifact, DiskArtifact, Snapshot, SnapshotType
+import host_tools.logging as log_tools
+import framework.utils as utils
 
 
 class MicrovmBuilder:
     """Build fresh microvms or restore from snapshot."""
 
-    def __init__(self, root_path, bin_cloner_path):
+    def __init__(self, bin_cloner_path):
         """Initialize microvm root and cloning binary."""
-        self.root_path = root_path
         self.bin_cloner_path = bin_cloner_path
+        self.init_root_path()
 
-    def build(self, kernel: Artifact, disks: [DiskArtifact], config: Artifact):
+    def init_root_path(self):
+        """Initialize microvm root path."""
+        self.root_path = tempfile.mkdtemp(prefix="fctest-")
+
+    def build(self,
+              kernel: Artifact,
+              disks: [DiskArtifact],
+              ssh_key: Artifact,
+              config: Artifact,
+              enable_diff_snapshots=False):
         """Build a fresh microvm."""
+        self.init_root_path()
+
         vm = init_microvm(self.root_path, self.bin_cloner_path)
-        vm.setup()
+        vm.jailer.daemonize = False
+
+        vm.jailer.extra_args.update({'seccomp-level': '0'})
```

Comment:
Is `seccomp-level 0` needed?

Reply:
Have to check if it works without, last time I was running these tests there was at least one ioctl that is not allowed by seccomp filter.

Reply:
Seems to be fine now.